CRM/Contact/Form/Search/Criteria.php

```diff
@@ -539,4 +541,19 @@ public static function CiviCase(&$form) {
     CRM_Case_BAO_Query::buildSearchForm($form);
   }
 
+  public static function getEnabledComponents() {
+    $apiResult = civicrm_api3('Setting', 'Get', array());
+    return $apiResult['values'][1]['enable_components'];
+  }
+
+  //the idea is to validate everything in here and remove based on possible situations
+  public static function validatePossibleComponents(&$componentModes) {
+    $enabledComponents = self::getEnabledComponents();
```

Comment:
Yes, and also instead of defining and calling new function we can make use of existing function for the same purpose, like 
```
$enabledComponents = self::getEnabledComponents();
if (!array_key_exists('CiviMail', $enabledComponents)) {
...
```
So can you update your PR with desired changes? Will test and get it merged afterward.